frb_macros/src/lib.rs

```diff
@@ -10,25 +10,48 @@ use proc_macro::*;
 // frb-coverage:ignore-start
 #[proc_macro_attribute]
 pub fn frb(attribute: TokenStream, item: TokenStream) -> TokenStream {
-    let item_str = item.to_string();
-    let mut new_str = String::with_capacity(item_str.len());
-    let mut last_end = 0;
-    while let Some(start) = item_str[last_end..].find("#[frb(") {
-        let start = start + last_end;
-        let end = item_str[start..].find(")]").unwrap() + start + 2;
-        new_str.push_str(&item_str[last_end..start]);
-        new_str.push_str("\n/// frb_marker: ");
-        new_str.push_str(&item_str[start..end]);
-        new_str.push('\n');
-        last_end = end;
-    }
-    new_str.push_str(&item_str[last_end..]);
-    let item: TokenStream = new_str.parse().unwrap();
+    let mut attribute = format_frb_attribute(format!("#[frb({attribute})]"));
+    let item = strip_frb_attr(item);
+    attribute.extend(item);
+    attribute
```

Comment:
nit:
```suggestion
    [format_frb_attribute(format!("#[frb({attribute})]")), strip_frb_attr(item)].concat()
```

Not sure, but I guess after `attribute.extend(item);`, the `attribute` variable does not mean attributes, but mean attributes *and* body?